core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -499,6 +504,20 @@ void finalize() {
 
     Experimental::pause_tools();
   }
+#ifdef KOKKOS_ENABLE_TUNING
+  // clean up string candidate set
+  for (auto& metadata_pair : Experimental::variable_metadata) {
+    auto metadata = metadata_pair.second;
+    if ((metadata.type == Experimental::ValueType::kokkos_value_string) &&
+        (metadata.valueQuantity ==
+         Experimental::CandidateValueType::kokkos_value_set)) {
+      auto candidate_set = metadata.candidates.set;
+      for (size_t index = 0; index < candidate_set.size; ++index) {
+        free(candidate_set.values.string_value[index]);
```

Comment:
These were allocated with `new` not `malloc` so you've got to use `delete`

Reply:
Thanks, I never use `new` so I forgot about it. Good catch, that appears to be it